pkg/controller/sync/version/cluster.go

```diff
@@ -17,35 +17,28 @@ limitations under the License.
 package version
 
 import (
-	"context"
-
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	pkgruntime "k8s.io/apimachinery/pkg/runtime"
 
 	fedv1a1 "github.com/kubernetes-sigs/federation-v2/pkg/apis/core/v1alpha1"
-	genericclient "github.com/kubernetes-sigs/federation-v2/pkg/client/generic"
 	"github.com/kubernetes-sigs/federation-v2/pkg/controller/util"
 )
 
-type clusterVersionAdapter struct {
-	client genericclient.Client
-}
+type clusterVersionAdapter struct{}
 
-func newClusterVersionAdapter(client genericclient.Client) VersionAdapter {
-	return &clusterVersionAdapter{client}
+func (*clusterVersionAdapter) TypeName() string {
+	return "ClusterPropagatedVersion"
```

Comment:
Does it make sense to use `reflect.TypeOf(&fedv1a1.ClusterPropagatedVersion{}).Name()` rather then hard coding the string here.

Reply:
Your suggestion is functionally equivalent, but I think it's out of scope here.  I'm not actually changing this particular code, though the diff may appear otherwise due to all the deletions.  I'm also not inclined to optimize the version stuff since there's a good chance it will be changing in the near future (e.g. to support selector-based overrides).

Reply:
Ok.